Packs/TrendMicroEmailSecurity/Integrations/TrendMicroEmailSecurityEventCollector/TrendMicroEmailSecurityEventCollector_test.py

```diff
@@ -0,0 +1,356 @@
+import pytest
+from CommonServerPython import *  # noqa # pylint: disable=unused-wildcard-import
+from TrendMicroEmailSecurityEventCollector import (
+    Client,
+    order_first_fetch,
+    managing_set_last_run,
+    fetch_by_event_type,
+    fetch_events_command,
+    remove_sensitive_from_events,
+    NoContentException,
+    DATE_FORMAT_EVENT,
+)
+
+
+@pytest.fixture()
+def mock_client() -> Client:
+    return Client(
+        base_url="test", username="test", api_key="test", verify=False, proxy=False
+    )
+
+
+@pytest.mark.parametrize("first_fetch", [("3 days"), ("32 hours")])
+def test_order_first_fetch(first_fetch: str):
+    assert order_first_fetch(first_fetch)
+
+
+@pytest.mark.parametrize("first_fetch", [("7 days"), ("4321 minutes")])
+def test_order_first_fetch_failure(first_fetch: str):
+    with pytest.raises(
+        ValueError,
+        match="The request retrieves logs created within 72 hours at most before sending the request\n"
+        "Please put in the First Fetch Time parameter a value that is at most 72 hours / 3 days",
+    ):
+        order_first_fetch(first_fetch)
+
+
+def test_handle_error_no_content(mock_client: Client):
+    class Response:
+        status_code = 204
+
+    with pytest.raises(NoContentException, match="No content"):
+        mock_client.handle_error_no_content(Response())
+
+
+def test_handle_error_no_content_with_other_errors(mock_client: Client):
+    class Response:
+        status_code = 404
+        reason = "test"
+
+        def json(self):
+            return {}
+
+    with pytest.raises(DemistoException):
+        mock_client.handle_error_no_content(Response())
+
+
+@pytest.mark.parametrize(
+    "event, expected_results",
+    [
+        (
+            {
+                "subject": "test",
+                "attachments": [{"fileName": "test", "sha256": "test"}],
+            },
+            {"attachments": [{"sha256": "test"}]},
+        ),
+        ({"subject": "test", "attachments": []}, {"attachments": []}),
+        ({"subject": "test", "attachments": None}, {"attachments": None}),
+    ],
+)
+def test_remove_sensitive_from_events(event: dict, expected_results: dict):
+    remove_sensitive_from_events(event)
+    assert event == expected_results
+
+
+def test_generate_authorization_encoded(mock_client: Client):
+    authorization_encoded = mock_client.generate_authorization_encoded("test", "test_api_key")
+    assert authorization_encoded == "dGVzdDp0ZXN0X2FwaV9rZXk="
+
+
+@pytest.mark.parametrize(
```

Comment:
consider using `pytest.param` to give `id`s to tests, making it easier to debug in the future.